Packs/GenericSQL/Integrations/GenericSQL/GenericSQL.py

```diff
@@ -38,12 +38,35 @@ def __init__(self, dialect: str, host: str, username: str, password: str, port:
         self.password = password
         self.port = port
         self.dbname = database
-        self.connect_parameters = connect_parameters
+        self.connect_parameters = self.parse_connect_parameters(connect_parameters, dialect)
         self.ssl_connect = ssl_connect
         self.use_pool = use_pool
         self.pool_ttl = pool_ttl
         self.connection = self._create_engine_and_connect()
 
+    @staticmethod
+    def parse_connect_parameters(connect_parameters: str, dialect: str) -> dict:
```

Comment:
I am a bit concerned on how we are parsing the query part of the url (building our own regex). There is additional logic for query url parsing. I did a quick search and from what I see sqlalchemy uses internally: https://docs.python.org/3/library/urllib.parse.html#urllib.parse.parse_qsl to parse the query part. I think we should use this utility function.